pkg/restapi/edv/edverrors/edverrors.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edverrors
+
+import "fmt"
+
+const (
+	// VaultNotFoundErrMsg is the message returned by the EDV server when a vault can't be found.
+	VaultNotFoundErrMsg = "specified vault does not exist"
+	// DocumentNotFoundErrMsg is the message returned by the EDV server when a document can't be found
+	// within the given vault.
+	DocumentNotFoundErrMsg = "specified document does not exist"
+	// DuplicateVaultErrMsg is the message returned by the EDV server when an attempt is made to create a vault with
+	// the same ID as an already existing vault.
+	DuplicateVaultErrMsg = "vault already exists"
+	// DuplicateDocumentErrMsg is the message returned by the EDV server when an attempt is made to create a document with
+	// the same ID as an already existing document inside the vault.
+	DuplicateDocumentErrMsg = "a document with the given ID already exists"
+	// NotBase58EncodedErrMsg is the message returned by the EDV server when an attempt is made
+	// to create a document with an ID that is not a base58-encoded value (which is required by the EDV spec).
+	NotBase58EncodedErrMsg = "document ID must be a base58-encoded value"
+	// Not128BitValueErrMsg is the message returned by the EDV server when an attempt is made
+	// to create a document with an ID that is base58-encoded, but the original value was not 128 bits long
+	// (which is required by the EDV spec).
+	Not128BitValueErrMsg = "document ID is base58-encoded, but original value before encoding was not 128 bits long"
+	// VaultQueryFailureToWriteFailureResponseErrMsg is used when an unexpected failure happens while the response is
+	// being written after a failure occurs while querying a vault.
+	VaultQueryFailureToWriteFailureResponseErrMsg = "Failed to write response for vault query failure: %s"
+	// VaultQueryFailureToWriteSuccessResponseErrMsg is used when an unexpected failure happens while the response is
+	// being written after successfully querying a vault.
+	VaultQueryFailureToWriteSuccessResponseErrMsg = "Failed to write response for vault query success: %s"
+)
+
+// ErrVaultNotFound is used when a vault could not be found in the provider.
+var ErrVaultNotFound = fmt.Errorf("%s", VaultNotFoundErrMsg)
+
+// ErrDocumentNotFound is used when a document could not be found in a vault.
```

Comment:
If you want to make an error as a constant you can do the following:
```
const ErrVaultNotFound  = edvError("specified vault does not exist")

// edvError defines error
type edvError string

// Error satisfies build-in error interface
func (e edvError) Error() string { return string(e) }

```

Reply:
@soluchok Thanks for the tip! I've implemented it in my code. It's make the edverrors.go file much shorter.